src/_zkapauthorizer/tahoe.py

```diff
@@ -300,6 +332,9 @@ def download(self, outpath, cap, child_path):
     def upload(self, inpath):
         return upload(self.client, inpath, self._api_root)
 
+    def upload_bytes(self, data):
+        return upload_bytes(self.client, data, self._api_root)
```

Comment:
This should have a new test or two added to `_zkapauthorizer.tests.test_tahoe`, probably on `UploadDownloadTestsMixin` - and then the test double will need an implementation too.

Reply:
Does it, though? It's already tested via `upload()` -- since it's basically just the guts of it. (I guess I'm "hoping" that we only need one of these, not both...)

Reply:
It looked to me like the test double didn't have an implementation of `upload_bytes` at all - so any test code that uses the test double will break if it tries to call that method?

Also I don't think I object to the path-based upload going away; the new one seems better.